homeassistant/components/bme280/sensor.py

```diff
@@ -1,166 +1,164 @@
+                )
             )
+    async_add_entities(entities, True)
 
-    async_add_entities(dev, True)
-
-
-class BME280Handler:
-    """BME280 sensor working in i2C bus."""
 
-    def __init__(self, sensor):
-        """Initialize the sensor handler."""
-        self.sensor = sensor
-        self.update(True)
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self, first_reading=False):
-        """Read sensor data."""
-        self.sensor.update(first_reading)
-
-
-class BME280Sensor(SensorEntity):
+class BME280Sensor(CoordinatorEntity, SensorEntity):
     """Implementation of the BME280 sensor."""
 
-    def __init__(self, bme280_client, sensor_type, temp_unit, name):
+    def __init__(self, sensor_type, temp_unit, name, coordinator):
         """Initialize the sensor."""
+        super().__init__(coordinator)
         self._attr_name = f"{name} {SENSOR_TYPES[sensor_type][0]}"
-        self.bme280_client = bme280_client
         self.temp_unit = temp_unit
         self.type = sensor_type
         self._attr_unit_of_measurement = SENSOR_TYPES[sensor_type][1]
         self._attr_device_class = SENSOR_TYPES[sensor_type][2]
 
-    async def async_update(self):
-        """Get the latest data from the BME280 and update the states."""
-        await self.hass.async_add_executor_job(self.bme280_client.update)
-        if self.bme280_client.sensor.sample_ok:
-            if self.type == SENSOR_TEMP:
-                if self.temp_unit == TEMP_FAHRENHEIT:
-                    self._attr_state = round(celsius_to_fahrenheit(self.state), 2)
-                else:
-                    self._attr_state = round(self.bme280_client.sensor.temperature, 2)
-            elif self.type == SENSOR_HUMID:
-                self._attr_state = round(self.bme280_client.sensor.humidity, 1)
-            elif self.type == SENSOR_PRESS:
-                self._attr_state = round(self.bme280_client.sensor.pressure, 1)
-        else:
-            _LOGGER.warning("Bad update of sensor.%s", self.name)
+    @callback
+    def update_from_latest_data(self) -> None:
+        """Update the entity from the latest data."""
+        if self.type == SENSOR_TEMP:
+            temperature = round(self.coordinator.data.temperature, 1)
+            if self.temp_unit == TEMP_FAHRENHEIT:
+                temperature = round(celsius_to_fahrenheit(temperature), 1)
+            self._attr_state = temperature
+        elif self.type == SENSOR_HUMID:
+            self._attr_state = round(self.coordinator.data.humidity, 1)
+        elif self.type == SENSOR_PRESS:
+            self._attr_state = round(self.coordinator.data.pressure, 1)
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+
+        @callback
+        def update() -> None:
+            """Update the state."""
+            self.update_from_latest_data()
+            self.async_write_ha_state()
+
+        self.async_on_remove(self.coordinator.async_add_listener(update))
```

Comment:
Why did we add this? The coordinator entity already handles this.

Reply:
I resolve the merge conflict. The "state" method has been replaced by _attr_state. How can I update the state in a different way?

Reply:
Keep the `state` property and remove the callback listener here.